### PR TITLE
WIP - API Rework (MK Dev)

### PR DESCRIPTION
**Pull request by kevzlou7979**
- Stepper
 - CompleteEvent implementation
 - Start, Next and Previous implementation
 - Animation transition upon navigating steps
- JsCamera fixed license header 
- Bubble Clean and fixed intialization problem (Support with Errai).
- Cutout Added missing Open Event.
- Bubble
	-	Clean and fixed intialization problem (Support with Errai)			
- Cutout 
	- Added missing Open Event.			6ea3da2
- Changed Runnable param to Functions.Func (JSInterop + Lambda).			
- ComboBox
	 - Fixed the selection / value change event.
- MaterialTimePicker - Fixed onLoad() twice (Support with Errai)
- Implement AddinsCssName Constants to clean up each predefined css for all individual widgets.

### CODE REVIEW

File: src/main/java/gwt/material/design/addins/client/stepper/events/StartEvent.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+package gwt.material.design.addins.client.stepper.events;
+
+
+import com.google.gwt.event.shared.EventHandler;
+import com.google.gwt.event.shared.GwtEvent;
+import com.google.gwt.event.shared.HasHandlers;
+
+public class StartEvent extends GwtEvent<StartEvent.StartHandler> {
+
+
```
Comment (BenDol) [inline src/main/java/gwt/material/design/addins/client/stepper/events/StartEvent.java]: remove this new line

File: src/main/java/gwt/material/design/addins/client/stepper/events/PreviousEvent.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package gwt.material.design.addins.client.stepper.events;
+
+
```
Comment (BenDol) [inline src/main/java/gwt/material/design/addins/client/stepper/events/PreviousEvent.java]: remove this new line

File: src/main/java/gwt/material/design/addins/client/stepper/events/PreviousEvent.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import com.google.gwt.event.shared.EventHandler;
+import com.google.gwt.event.shared.GwtEvent;
+import com.google.gwt.event.shared.HasHandlers;
+
+public class PreviousEvent extends GwtEvent<PreviousEvent.PreviousHandler> {
+
+
```
Comment (BenDol) [inline src/main/java/gwt/material/design/addins/client/stepper/events/PreviousEvent.java]: remove this new line

File: src/main/java/gwt/material/design/addins/client/stepper/events/StartEvent.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package gwt.material.design.addins.client.stepper.events;
+
+
```
Comment (BenDol) [inline src/main/java/gwt/material/design/addins/client/stepper/events/StartEvent.java]: remove this new line

File: src/main/java/gwt/material/design/addins/client/stepper/events/NextEvent.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+package gwt.material.design.addins.client.stepper.events;
+
+
+import com.google.gwt.event.shared.EventHandler;
+import com.google.gwt.event.shared.GwtEvent;
+import com.google.gwt.event.shared.HasHandlers;
+
+public class NextEvent extends GwtEvent<NextEvent.NextHandler> {
+
+
```
Comment (BenDol) [inline src/main/java/gwt/material/design/addins/client/stepper/events/NextEvent.java]: remove this new line

File: src/main/java/gwt/material/design/addins/client/stepper/events/CompleteEvent.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package gwt.material.design.addins.client.stepper.events;
+
+
```
Comment (BenDol) [inline src/main/java/gwt/material/design/addins/client/stepper/events/CompleteEvent.java]: remove this new line

File: src/main/java/gwt/material/design/addins/client/stepper/base/HasStepHandler.java [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+package gwt.material.design.addins.client.stepper.base;
+
+import com.google.gwt.event.shared.HandlerRegistration;
+import com.google.gwt.event.shared.HasHandlers;
+import gwt.material.design.addins.client.stepper.events.CompleteEvent;
+import gwt.material.design.addins.client.stepper.events.NextEvent;
+import gwt.material.design.addins.client.stepper.events.PreviousEvent;
+import gwt.material.design.addins.client.stepper.events.StartEvent;
+
+public interface HasStepHandler extends HasHandlers {
```
Comment (BenDol) [inline src/main/java/gwt/material/design/addins/client/stepper/base/HasStepHandler.java]: This should be plural `HasStepHandlers`

File: src/main/java/gwt/material/design/addins/client/stepper/events/CompleteEvent.java (hunk 6 above)
Comment (kevzlou7979) [inline src/main/java/gwt/material/design/addins/client/stepper/events/CompleteEvent.java]: Ok corrected

File: src/main/java/gwt/material/design/addins/client/stepper/events/StartEvent.java (hunk 4 above)
Comment (kevzlou7979) [inline src/main/java/gwt/material/design/addins/client/stepper/events/StartEvent.java]: Ok corrected

File: src/main/java/gwt/material/design/addins/client/stepper/events/PreviousEvent.java (hunk 2 above)
Comment (kevzlou7979) [inline src/main/java/gwt/material/design/addins/client/stepper/events/PreviousEvent.java]: Ok corrected

**Pull request merged**